tests/checkers/unittest_imports.py

```diff
@@ -138,6 +138,55 @@ def test_preferred_module(capsys: CaptureFixture[str]) -> None:
         # assert there were no errors
         assert len(errors) == 0
 
+        # test preferred-modules trigger case with submodules
+        Run(
+            [
+                f"{os.path.join(REGR_DATA, 'preferred_module/unpreferred_submodule.py')}",
+                "-d all",
+                "-e preferred-module",
+                # prefer os.path instead of pathlib (for triggering test)
+                "--preferred-modules=os.path:pathlib",
+            ],
+            exit=False,
+        )
+        output, errors = capsys.readouterr()
+
+        # test preferred-modules ignore case with submodules
+        Run(
+            [
+                f"{os.path.join(REGR_DATA, 'preferred_module/unpreferred_submodule.py')}",
+                "-d all",
+                "-e preferred-module",
+                # prefer pathlib instead of os.stat (for untriggered test)
+                "--preferred-modules=os.stat:pathlib",
+            ],
+            exit=False,
+        )
+        output, errors = capsys.readouterr()
+
+        # assert that we didn't see preferred-modules triggered
+        assert "Your code has been rated at 10.00/10" in output
+        # assert there were no errors
+        assert len(errors) == 0
+
+        # test preferred-modules base module for implemented submodule (from ... import)
+        Run(
+            [
+                f"{os.path.join(REGR_DATA, 'preferred_module/unpreferred_submodule.py')}",
+                "-d all",
+                "-e preferred-module",
+                # prefer pathlib instead of os (for triggering test)
+                "--preferred-modules=os:pathlib",
+            ],
+            exit=False,
+        )
+        output, errors = capsys.readouterr()
+
+        # assert that we saw preferred-modules triggered with base module
+        assert "Prefer importing 'pathlib' instead of 'os'" in output
+        # assert there were no errors
+        assert len(errors) == 0
+
```

Comment:
This could be a small functional tests implemented in 10 lines with exact match and automated upgrade if something was to change. @DanielNoord are those unit test really faster or should we move everything to functional ?

Reply:
In this case the unit tests aren't much faster, in general they probably are because there is less overhead.

Anyway, I don't really mind these as they built upon the previously existing test but I understand your point.